### ISSUE TITLE
Build in Master and 6.x is Broken after merging in X-Pack.

### ISSUE THREAD

**Issue by original-brownbear** [elastic/logstash#9478]
Many failures coming from settings registration in some way it seems.

Example `runner_spec.rb`:

```sh
  18) LogStash::Runner logstash modules --modules with an unavailable module specified should log fatally and return a bad exit code
      Failure/Error:
        expect(hash).to be_a_config_loading_error_hash(
          /The modules specified are not available yet. Specified modules: \["fancypants"\] Available modules:/)
      
        expected {:error=>#<ArgumentError: Setting "xpack.management.enabled" has already been registered as #<LogStas...core/runner.rb:71:in `run'", "/Users/brownbear/src/logstash/lib/bootstrap/rspec.rb:13:in `<main>'"]} to be a config loading error hash /The modules specified are not available yet. Specified modules: \["fancypants"\] Available modules:/
      # ./logstash-core/spec/logstash/runner_spec.rb:408:in `block in (root)'
      # ./logstash-core/lib/logstash/runner.rb:386:in `execute'
      # ./vendor/bundle/jruby/2.3.0/gems/clamp-0.6.5/lib/clamp/command.rb:67:in `run'
      # ./logstash-core/lib/logstash/runner.rb:230:in `run'
      # ./vendor/bundle/jruby/2.3.0/gems/clamp-0.6.5/lib/clamp/command.rb:132:in `run'
      # ./logstash-core/spec/logstash/runner_spec.rb:418:in `block in (root)'
      # ./vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block in /Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb'
      # ./lib/bootstrap/rspec.rb:13:in `<main>'
      # ------------------
      # --- Caused by: ---
      # ArgumentError:
      #   Setting "xpack.management.enabled" has already been registered as #<LogStash::Setting::Boolean:0x298e5201 @name="xpack.management.enabled", @value=nil, @value_is_set=false, @klass=Object, @default=false, @validator_proc=nil>
      #   ./logstash-core/lib/logstash/settings.rb:25:in `register'

  19) LogStash::Runner logstash modules --modules with an available module specified but no connection to elasticsearch should log fatally and return a bad exit code
      Failure/Error:
        expect(hash).to be_a_config_loading_error_hash(
          /Failed to import module configurations to Elasticsearch and\/or Kibana. Module: tester has/)
      
        expected {:error=>#<ArgumentError: Setting "xpack.management.enabled" has already been registered as #<LogStas...core/runner.rb:71:in `run'", "/Users/brownbear/src/logstash/lib/bootstrap/rspec.rb:13:in `<main>'"]} to be a config loading error hash /Failed to import module configurations to Elasticsearch and\/or Kibana. Module: tester has/
      # ./logstash-core/spec/logstash/runner_spec.rb:349:in `block in (root)'
      # ./logstash-core/lib/logstash/runner.rb:386:in `execute'
      # ./vendor/bundle/jruby/2.3.0/gems/clamp-0.6.5/lib/clamp/command.rb:67:in `run'
      # ./logstash-core/lib/logstash/runner.rb:230:in `run'
      # ./vendor/bundle/jruby/2.3.0/gems/clamp-0.6.5/lib/clamp/command.rb:132:in `run'
      # ./logstash-core/spec/logstash/runner_spec.rb:359:in `block in (root)'
      # ./vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block in /Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb'
      # ./lib/bootstrap/rspec.rb:13:in `<main>'
      # ------------------
      # --- Caused by: ---
      # ArgumentError:
      #   Setting "xpack.management.enabled" has already been registered as #<LogStash::Setting::Boolean:0x6960a39f @name="xpack.management.enabled", @value=nil, @value_is_set=false, @klass=Object, @default=false, @validator_proc=nil>
      #   ./logstash-core/lib/logstash/settings.rb:25:in `register'

  20) LogStash::Runner path.settings if does not exist should not terminate logstash
      Failure/Error: expect(subject.run(args)).to eq(0)
      
        expected: 0
             got: 1
      
        (compared using ==)
      # ./logstash-core/spec/logstash/runner_spec.rb:508:in `block in (root)'
      # ./vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block in /Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb'
      # ./lib/bootstrap/rspec.rb:13:in `<main>'

Finished in 5.47 seconds (files took 5.13 seconds to load)
38 examples, 20 failures

Failed examples:

```

=> I'm on it :)

**Comment by original-brownbear** [comment]
The strange thing here is that this seems to happen only on my local, not on Jenkins. See fix #9480 though, it's fairly obvious how repeated invocations of `runner#execute` should trigger this imo,

**Comment by original-brownbear** [comment]
Fixed via https://github.com/elastic/logstash/pull/9480
